src/main/scala/satify/update/solver/dpll/Dpll.scala

```diff
@@ -0,0 +1,79 @@
+package satify.update.solver.dpll
+
+import satify.model.Bool.{False, True}
+import satify.model.CNF.{And, Not, Or, Symbol}
+import satify.model.{CNF, Variable}
+import satify.model.dpll.{Constraint, Decision, DecisionTree, PartialModel}
+import satify.model.dpll.DecisionTree.{Branch, Leaf}
+import satify.update.solver.dpll.cnf.CNFSimplification.simplifyCnf
+import satify.update.solver.dpll.cnf.CNFSat.{isSat, isUnsat}
+import satify.update.solver.dpll.Optimizations.{pureLit, uProp}
+import satify.update.solver.dpll.utils.PartialModelUtils.*
+
+import java.util.concurrent.Executors
+import scala.annotation.tailrec
+import scala.util.Random
+
+object Dpll:
+
+  private val rnd = Random(64)
+
+  def dpll(dec: Decision): DecisionTree =
+
+    case class Frame(d: Decision, done: List[DecisionTree], todos: List[Decision])
+
+    @tailrec
+    def step(stack: List[Frame]): DecisionTree = stack match
+      case Frame(d, done, Nil) :: tail =>
+        val ret =
+          done match
+            case ::(head, next) => Branch(d, next.head, head)
+            case Nil => Leaf(d)
+        tail match
+          case Nil => ret
+          case Frame(tn, td, tt) :: more =>
+            step(Frame(tn, ret :: td, tt) :: more)
+      case Frame(d, done, x :: xs) :: tail =>
+        if isUnsat(d.cnf) || isSat(d.cnf) then step(Frame(d, Nil, Nil) :: tail)
+        else step(Frame(x, Nil, decide(x)) :: Frame(d, done, xs) :: tail)
+      case Nil => throw new Error("Shouldn't happen")
+
+    step(List(Frame(dec, Nil, decide(dec))))
+
+  private def decide(d: Decision): List[Decision] = d match
+    case Decision(_, cnf) =>
+      uProp(cnf) match
+        case Some(c) => uPropDec(d, c)
+        case _ =>
+          pureLit(d) match
+            case Some(c) => pureLitDec(d, c)
+            case _ => rndDec(d)
+
+  private def rndDec(d: Decision): List[Decision] = d match
```

Comment:
I would use short and concise names mainly on math stuffs, otherwise more explicative names are better.